front_end/panels/timeline/TimelinePanel.ts

```diff
@@ -712,6 +712,14 @@ export class TimelinePanel extends UI.Panel.Panel implements Client, TimelineMod
         SDK.ReactNativeApplicationModel.ReactNativeApplicationModel,
         {
           modelAdded: (model: SDK.ReactNativeApplicationModel.ReactNativeApplicationModel) => {
+            model.addEventListener(
+              SDK.ReactNativeApplicationModel.Events.METADATA_UPDATED,
+              (event: Common.EventTarget.EventTargetEvent<Protocol.ReactNativeApplication.MetadataUpdatedEvent>) => {
+                if (event.data.platform === 'android') {
+                  this.showScreenshotsToolbarCheckbox?.setVisible(true);
+                }
+              }
```

Comment:
Could there be a race? This event could be emitted before TimelinePanel is loaded. You can try reproducing this by navigating to Console panel, for example, and then re-open DevTools instance. It will load Console panel first, and will only load Timeline when user navigates to Performance tab.

Reply:
You could follow the approach that I used for fetching the app name as a trace title - https://github.com/facebook/react-native-devtools-frontend/pull/207

This would work if the model was already loaded. If not, your approach with model listener should cover the other case.